gdnative-derive/src/methods.rs

```diff
@@ -79,28 +82,36 @@ pub(crate) fn derive_methods(item_impl: ItemImpl) -> TokenStream2 {
     let methods = export
         .methods
         .into_iter()
-        .map(|ExportMethod { sig, args }| {
+        .map(|ExportMethod { sig, export_args , optional_args, exist_base_arg}| {
             let sig_span = sig.ident.span();
 
             let name = sig.ident;
-            let name_string = args.name_override.unwrap_or_else(|| name.to_string());
+            let name_string = export_args.name_override.unwrap_or_else(|| name.to_string());
             let ret_span = sig.output.span();
-            let ret_ty = match sig.output {
+            let ret_ty = match &sig.output {
                 syn::ReturnType::Default => quote_spanned!(ret_span => ()),
                 syn::ReturnType::Type(_, ty) => quote_spanned!( ret_span => #ty ),
             };
 
             let arg_count = sig.inputs.len();
 
-            if arg_count < 2 {
+            if arg_count == 0 {
+                return syn::Error::new(
+                    sig_span,
+                    "exported methods must take self as argument",
```

Comment:
maybe: "exported methods cannot be associated functions"

because they don't literally take `self` but also e.g. `&mut self`.

Reply:
"Associated functions" is a term that includes "methods" and should not be described as such. ([reference](https://doc.rust-lang.org/reference/items/implementations.html#inherent-implementations))

The [reference](https://doc.rust-lang.org/reference/items/associated-items.html#methods) also uses the term "self parameter". So, change the terms from argument to parameter.

Reply:
Thanks for the correction, you're right! "self parameter" is fine with me 👍